infrastructure/constructs/notify.py

```diff
@@ -0,0 +1,73 @@
+from os import environ
+
+from aws_cdk import (
+    aws_events,
+    aws_events_targets,
+    aws_lambda,
+    aws_lambda_python,
+    aws_sns,
+    aws_sns_subscriptions,
+)
+from aws_cdk.core import Construct
+
+from backend.notify_slack.task import SLACK_NOTIFY_URL_ENV_NAME
+
+from .bundled_code import bundled_code
+from .lambda_config import LAMBDA_TIMEOUT, PYTHON_RUNTIME
+
+
+class Notify(Construct):
+    def __init__(
+        self,
+        scope: Construct,
+        stack_id: str,
+        *,
+        botocore_lambda_layer: aws_lambda_python.PythonLayerVersion,
+        env_name: str,
+        state_machine_arn: str,
+    ) -> None:
+        super().__init__(scope, stack_id)
+
+        slack_notify_function = aws_lambda.Function(
+            scope,
+            f"{env_name}-notify-slack-function",
+            function_name=f"{env_name}-notify-slack",
+            handler="notify_slack.task.lambda_handler",
+            runtime=PYTHON_RUNTIME,
+            timeout=LAMBDA_TIMEOUT,
+            code=bundled_code("notify_slack"),
+            layers=[botocore_lambda_layer],  # type: ignore[list-item]
+        )
+        if SLACK_NOTIFY_URL_ENV_NAME in environ:
+            slack_notify_function.add_environment(
+                SLACK_NOTIFY_URL_ENV_NAME,
+                environ.get(SLACK_NOTIFY_URL_ENV_NAME),  # type: ignore[arg-type]
```

Comment:
You know the environment variable exists here, so you can use `environ[SLACK_NOTIFY_URL_ENV_NAME]`. This also makes the value no longer `Optional`, so you can probably remove the annotation as well.